### PR TITLE
`d2l-rubric-levels-editor` conversion - US132993

### PR DESCRIPTION
**Pull request by imcdonald-d2l**
https://rally1.rallydev.com/#/57732444928d/custom/373260458992?detail=%2Fuserstory%2F612451712579

### CODE REVIEW

File: src/mixins/telemetry-mixin.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+
+const TelemetryMixinInternal = (superclass) => class extends superclass {
```
Comment (imcdonald-d2l) [inline src/mixins/telemetry-mixin.js:3]: Just converting existing [telemetry behavior](https://github.com/Brightspace/d2l-rubric/blob/master/src/telemetry-behavior.js) into a mixin.

File: src/components/editor/rubric-levels-editor.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+			</div>
+			<div class="gutter-right"></div>
+		`;
+	}
+
+	_checkSize() {
+		requestAnimationFrame(() => {
+			const section = this.shadowRoot.querySelector('#levels-section');
+			if (section && section.offsetWidth) {
+				this.dispatchEvent(new CustomEvent('d2l-rubric-editor-levels-width-changed', {
```
Comment (imcdonald-d2l) [inline src/components/editor/rubric-levels-editor.js:184]: This is for the existing `d2l-rubric-criteria-editor`. Listener is in the criteria group editor [here](https://github.com/Brightspace/d2l-rubric/blob/master/src/editor/d2l-rubric-criteria-group-editor.js#L353).  This is certainly not an ideal way to approach styling  but didn't want to re-write the css in a bunch of files to get around this. The CSS should be fixed from the bottom up starting with the criterion editor [here](https://github.com/Brightspace/d2l-rubric/blob/master/src/editor/d2l-rubric-criterion-editor.js#L644).

File: src/components/editor/rubric-levels-editor.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+	_renderLevels() {
+		const levelTemplates = [];
+		for (let i = 0; i < this._levels.length; i++) {
+			const level = this._levels[i];
+			levelTemplates.push(html`
+				<div class="cell" ?is-holistic="${this.isHolistic}">
+					<d2l-rubric-level-editor
+						.href="${level.selfLink}"
+						.token="${this.token}"
+						position-number="${this._getPositionNumber(i)}"
```
Comment (MitchelPaulin) [inline src/components/editor/rubric-levels-editor.js:339]: Might be better just to replace this with `i + 1`

Comment (imcdonald-d2l) [reply src/components/editor/rubric-levels-editor.js:339]: Yeah, agreed, uses an unnecessary function. 👍🏻

**Comment by imcdonald-d2l** [comment]
> Visual diff tests failed - pull request #1294 has been opened with the updated goldens.

Fixes some border styles that were previously messed up! 🎉

File: src/mixins/telemetry-mixin.js [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+			.setType(isError ? 'ErrorEvent' : this.eventType)
+			.setSourceId(this.sourceId)
+			.setBody(eventBody);
+
+		client.logUserEvent(event);
+		return event;
+	}
+
+	_markExists(markName) {
+		return window.performance.getEntriesByName(markName, 'mark').length > 0 ? true : false;
```
Comment (MitchelPaulin) [inline src/mixins/telemetry-mixin.js:298]:
```suggestion
		return window.performance.getEntriesByName(markName, 'mark').length > 0;
```

**Review by MitchelPaulin — APPROVED** [review]

**Pull request merged**